### ISSUE TITLE
High Package Activation Time

### ISSUE THREAD

**Issue by wottpal** [AtomLinter/linter-php#147]
Hi,
just wondering because in Timecop linter-php is nearly always the package with the highest activation time :/ Are there any obvious optimizations which could be made? For example the most times when I start Atom I don't even open a PHP file but nonetheless the package takes a while to activate..

<img width="117" alt="bildschirmfoto 2016-03-14 um 14 42 14" src="https://cloud.githubusercontent.com/assets/3930150/13746342/418544ee-e9f3-11e5-9382-c3aba4bc8e6c.png">

**Comment by hitalos** [comment]
On my Mac, the timecop also shows higher loading time to linter-php.
<img width="239" alt="captura de tela 2016-03-17 as 23 50 15" src="https://cloud.githubusercontent.com/assets/1332910/13867170/44adfe72-ec9b-11e5-82cc-1e72a41a26c8.png">
On ubuntu 15.10, using the same packages, this doesn't happens.

**Comment by Arcanemagus** [comment]
That's pretty ridiculous for a startup time... there are a few things that can be done to improve it slightly, but something is seriously wrong if you are seeing an activation time of that long.

Do you have a custom shell config script? On OS X we do several things to try to get a sane environment for execution because Apple decided it would be a bright idea to strip all the important stuff from it for GUI apps... one of the things we do is load a shell up and pull the environment from there so if you have things attempting to load in a non-interactive shell that are only needed in an interactive shell you might want to wrap them in conditionals.

**Comment by Arcanemagus** [comment]
The major thing to improve activation time is to _not_, at least not until it is actually needed. That part is a duplicate of https://github.com/AtomLinter/linter-php/issues/86. There are a few minor things that can be done other than that, but at best they might shave a ms or two off...

**Comment by iovis9** [comment]
It's becoming a real problem for me. OS X 10.11.4
<img width="208" alt="screen shot 2016-05-12 at 19 27 03" src="https://cloud.githubusercontent.com/assets/6660202/15224059/bf4ef156-1877-11e6-8b7f-e192579a8aa0.png">

I'll gladly answer any questions about my environment.

**Comment by iovis9** [comment]
Does someone still care about this? It's still the package that takes the most time to load (10 times more than the second one)

**Comment by Arcanemagus** [comment]
@iovis9 The package currently _works_, and unfortunately the time I have to spare on this organization has lately been dedicated to packages that don't fall into that category.

There are a few things that can be done here, and I'd gladly review a PR from somebody that wants to take the time to figure them out:

* Get `activationHooks` working properly (#86)
* Make the version check async
  * Or remove it entirely? This was mainly put in place for OS X users who were using the incredibly out of date version on their system without realizing it. It's so old the output it produces isn't parseable.
* Lazy load `atom-linter`, this shouldn't be that big of a performance difference, but a few ms is a few ms.
